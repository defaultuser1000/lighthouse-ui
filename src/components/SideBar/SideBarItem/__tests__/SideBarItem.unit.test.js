import React from 'react';
import {shallow} from 'enzyme/build';
import {SideBarItem} from '../SideBarItem';

describe('SideBarItem', () => {

    test('renders empty SideBarItem', () => {
        const wrapper = shallow(
            <SideBarItem/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders highlighted SideBarItem that navigates to /settings', () => {
        const wrapper = shallow(
            <SideBarItem highlighted icon='settings' label='Settings'/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('render non-highlighted SideBarItem that navigates to /settings', () => {
        const wrapper = shallow(
            <SideBarItem icon='settings' label='Settings'/>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

