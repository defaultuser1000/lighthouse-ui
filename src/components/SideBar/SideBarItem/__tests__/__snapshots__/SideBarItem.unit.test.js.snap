// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem render non-highlighted SideBarItem that navigates to /settings 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="settings"
        size="large"
      />
    </span>
    <span>
      Settings
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem renders highlighted SideBarItem that navigates to /settings 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="settings"
        size="large"
      />
    </span>
    <span>
      Settings
    </span>
  </div>
</MenuItem>
`;
